src/editor/Core/Filesystem.re

```diff
@@ -263,55 +299,50 @@ let createOniConfiguration = (~configDir, ~file) => {
 
 let getPath = (dir, file) => return(Utility.join([dir, file]));
 
-let createConfigIfNecessary = (configDir, file) =>
-  Utility.join([configDir, file])
-  |> (
-    configPath =>
-      stat(configDir)
-      >>= (
-        fun
-        | Some(dirStats) =>
-          isDir(dirStats)
-          /\/= (
-            _ =>
-              mkdir(configDir, ())
-              >>= (() => createOniConfiguration(~file, ~configDir))
-          )
-          /* config directory already exists */
-          >>= (
-            () =>
-              createOniConfiguration(~configDir, ~file)
-              /\/= error("Error creating configuration files because: %s")
-              >>= (() => return(configPath))
-          )
-        | None =>
-          mkdir(configDir, ())
-          >>= (() => createOniConfiguration(~configDir, ~file))
-          >>= (() => return(configPath))
-      )
-  );
+let createConfigIfNecessary = (configDir, file) => {
+  open Let_syntax;
+
+  let filepath = Utility.join([configDir, file]);
+
+  let%map dirStats = stat(configDir);
+  switch%bind (dirStats) {
+  | Some(stats) =>
+    isDir(stats)
+    /\/= (
```

Comment:
This operator is similar to bind but rather it specifically responds to `Error(v)` types returned from operators so that the error handling can be sequenced along with the success handling with a lot nested `match` statements